feat: deployment simplification

lekhovitsky opened the pull request:
In this PR:
* `PoolQuotaKeeperV3` is now deployed inside `PoolV3`'s constructor, which allows to store it as immutable variable making interactions with the system cheaper
* All the contracts that can read `acl` and `contractsRegister` instead of accepting a constructor argument now do so, reducing the chances of misconfiguration
* Constructors now consistently revert with `ZeroAddressException` 
* Better validation in `USDT_Transfer` trait
* Silence warnings in tests

lekhovitsky committed:
feat: simplify constructors

lekhovitsky committed:
test: silence warnings

lekhovitsky committed:
chore: deps cleanup

lekhovitsky committed:
feat: deploy quota keeper in pool constructor

It makes interactions much cheaper as now quota keeper is immutable in all contracts.
Besides, it simplifies the deployment process.

0xmikko commented on contracts/credit/CreditConfiguratorV3.sol:
```diff
@@ -62,10 +63,9 @@ contract CreditConfiguratorV3 is ICreditConfiguratorV3, ControlledTrait, Reentra
     /// @param _creditManager Credit manager to connect to
     /// @dev Copies allowed adaprters from the currently connected configurator
-    constructor(address _acl, address _creditManager) ControlledTrait(_acl) {
+    constructor(address _creditManager) ControlledTrait(IPoolV3(CreditManagerV3(_creditManager).pool()).acl()) {
```
it's better to add sanity check for _creditManager is not zero address. We implemeted it in prev versions as:
_creditManager == address(0) ? address(0) : IPoolV3....

lekhovitsky committed:
fix: consistent constructor inputs sanitization

The rule is to only add `nonZeroAddress` for arguments that are not sanitized in any other way.

lekhovitsky committed:
fix: revert with clear error on zero address in constructor

lekhovitsky committed:
fix: better validation in `USDT_Transfer` constructor
